backend/src/main/java/ca/bc/gov/backendstartapi/endpoint/FavouriteActivityEndpoint.java

```diff
@@ -54,8 +56,13 @@ public FavouriteActivityEntity createUserActivity(
    */
   @GetMapping(produces = "application/json")
   @PreAuthorize("hasRole('user_read')")
-  public List<FavouriteActivityEntity> getUserActivities() {
-    return favouriteActivityService.getAllUserFavoriteActivities();
+  public ResponseEntity<List<FavouriteActivityEntity>> getUserActivities() {
+    List<FavouriteActivityEntity> activityEntities =
+        favouriteActivityService.getAllUserFavoriteActivities();
+    if (activityEntities.isEmpty()) {
+      return ResponseEntity.status(HttpStatus.NO_CONTENT).body(activityEntities);
```

Comment:
[The description of the 204 status code by the HTTP Working Group](https://httpwg.org/specs/rfc9110.html#status.204) doesn't cover a case like this, so its use isn't outright discouraged. It doesn't look like this a case intended for it, though: it mentions "successfully applying an action to a resource" throughout its definition and examples, which doesn't fit to the behaviour expected of a [safe method](https://httpwg.org/specs/rfc9110.html#safe.methods), that doesn't change the server's state.

Apart from that, there's a semantic consideration here I think we should follow here: like in Java, it's easier for everyone involved if we return an actual collection (an empty `List` or an empty array) when a collection is expected, instead of something else (`null` or nothing, in this case).